tests/test_apypie.py

```diff
@@ -172,5 +178,48 @@ def test_http_call_post_with_params(api, requests_mock):
         'Content-Type': 'application/json',
     }
     params = {'test': 'all the things'}
-    requests_mock.post('https://api.example.com/', text='{}', headers=expected_headers)
-    api.http_call('post', '/', params)
+    rmock = requests_mock.post('https://api.example.com/', json={'result': 'tracer'}, headers=expected_headers)
+    assert api.http_call('post', '/', params) == {'result': 'tracer'}
+    assert rmock.call_count == 1
+    assert set(expected_headers.items()) <= set(rmock.last_request.headers.items())
```

Comment:
What is this testing exactly?

The [documentation of requests_mock](https://requests-mock.readthedocs.io/en/latest/matching.html#request-headers) says "[a] dictionary of headers can be supplied such that the request will only match if the available headers also match", so `api.http_call` will only succeed if it passes at least `expected_headers` (otherwise a `requests_mock.exceptions.NoMockAddress` is raised).

So it kinda feels `requests_mock` already verified that we passed the right headers, why doing it again?

Reply:
Well, then that is clearly a (documentation?) bug in `requests_mock`. When i add another rouge header to the expected list, the test only fails in line 184. So i conclude, that requests_mock only matches headers, that are both 'expected' and part of the actual request.

Reply:
That's... interesting. As it works just fine for me:

```
api = <apypie.api.Api instance at 0x7f1594fad878>, requests_mock = <requests_mock.mocker.Mocker object at 0x7f1594efbdd0>

    def test_http_call_get_headers(api, requests_mock):
        headers = {'X-Apypie-Test': 'Awesome'}
        expected_headers = {
            'Accept': 'application/json;version=1',
            'X-Yadda': 'badabu',
        }
        expected_headers.update(headers)
        requests_mock.get('https://api.example.com/', request_headers=expected_headers, text='{}')
>       api.http_call('get', '/', headers=headers)

tests/test_apypie.py:117: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
apypie/api.py:185: in http_call
    request = self._session.request(http_method, full_path, **kwargs)
.tox/py27/lib/python2.7/site-packages/requests/sessions.py:533: in request
    resp = self.send(prep, **send_kwargs)
.tox/py27/lib/python2.7/site-packages/requests_mock/mocker.py:111: in _fake_send
    return _original_send(session, request, **kwargs)
.tox/py27/lib/python2.7/site-packages/requests/sessions.py:646: in send
    r = adapter.send(request, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <requests_mock.adapter.Adapter object at 0x7f1594efbf10>, request = <requests_mock.request._RequestObjectProxy object at 0x7f1594ee94d0>, kwargs = {'cert': None, 'proxies': OrderedDict(), 'stream': False, 'timeout': None, ...}
matcher = <requests_mock.adapter._Matcher object at 0x7f1594efbd90>, resp = None

    def send(self, request, **kwargs):
        request = _RequestObjectProxy(request,
                                      case_sensitive=self._case_sensitive,
                                      **kwargs)
        self._add_to_history(request)
    
        for matcher in reversed(self._matchers):
            try:
                resp = matcher(request)
            except Exception:
                request._matcher = weakref.ref(matcher)
                raise
    
            if resp is not None:
                request._matcher = weakref.ref(matcher)
                resp.connection = self
                return resp
    
>       raise exceptions.NoMockAddress(request)
E       NoMockAddress: No mock address: GET https://api.example.com/

.tox/py27/lib/python2.7/site-packages/requests_mock/adapter.py:250: NoMockAddress
```

Reply:
OK. I know what happens. See https://github.com/Apipie/apypie/pull/40 -- you need to use `request_headers`, not `headers`